app/controllers/blog/posts_controller.rb

```
@@ -0,0 +1,54 @@
+
+      # Category pills in the view are now static links; controller doesn't need to provide @categories or @tags for filtering.
+
+      render "blog/posts/index"
+    end
+
+    def show
+      if @post.nil? || !@post.published # Ensure post exists in manifest and is published
+        return render_404
+      end
+
+      render 'blog/posts/show'
```

RuboCop (1.75.5)

[convention] Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

(Style/StringLiterals)